src/Strategy/Maven/Pom.hs

```diff
@@ -64,19 +65,30 @@ data MavenLabel
   | MavenLabelOptional Text
   deriving (Eq, Ord, Show)
 
-toDependency :: MavenPackage -> Set MavenLabel -> Dependency
-toDependency (MavenPackage group artifact version) = foldr applyLabel start
+toDependency :: MavenPackage -> Set MavenLabel -> MavenDependency
+toDependency (MavenPackage group artifact version) = foldr applyLabelToMavenDep start
   where
-    start :: Dependency
-    start =
-      Dependency
-        { dependencyType = MavenType
-        , dependencyName = group <> ":" <> artifact
-        , dependencyVersion = CEq <$> version
-        , dependencyLocations = []
-        , dependencyEnvironments = mempty
-        , dependencyTags = Map.empty
-        }
+    start :: MavenDependency
+    start = do
+      let dep =
+            Dependency
+              { dependencyType = MavenType
+              , dependencyName = group <> ":" <> artifact
+              , dependencyVersion = CEq <$> version
+              , dependencyLocations = []
+              , dependencyEnvironments = mempty
+              , dependencyTags = Map.empty
+              }
+      MavenDependency dep (Set.fromList [])
+
+    applyLabelToMavenDep :: MavenLabel -> MavenDependency -> MavenDependency
+    applyLabelToMavenDep lbl mavenDep = do
+      case lbl of
+        MavenLabelScope scope ->
+          mavenDep{dependency = depWithLabel, dependencyScopes = Set.insert scope (dependencyScopes mavenDep)}
+        _ -> mavenDep{dependency = depWithLabel}
+      where
```

Comment:
[Prefer] You should avoid (as much as possible) of having multiple where clauses, for readability, 

```haskell
someMethod = ...
    where
        someCall = ... 
            where
                someOtherCall
```

Reply:
That's a good point. What are the cases in which having multiple where clauses is acceptable? Just curious bc I see it done [here](https://github.com/fossas/fossa-cli/blob/bcd1142515684b565ccc8f67f4e91d61cc2e6d5b/src/Strategy/Maven/Pom.hs#L98C1-L98C1)